README.md

```diff
@@ -6,4 +6,75 @@
 [![Stackage Nightly](http://stackage.org/package/cake-slayer/badge/nightly)](http://stackage.org/nightly/package/cake-slayer)
 [![MPL-2.0 license](https://img.shields.io/badge/license-MPL--2.0-blue.svg)](LICENSE)
 
-Architecture of Haskel backend applications
+`cake-slayer` (do not confuse with `cakes-layer`) is a modern and
+batteries-included framework for creating backend in Haskell for
+web-applications. It allows you to scaffold working and extensible project in
+minutes.
+
+## Overview
+
+The main goal of `cake-slayer` is to provide a backbone for your Haskell
+backend. Unlike many other Haskell libraries, `cake-slayer` doesn't try to be as
+abstract as possible. It includes best-practices and makes some architecture
+decisions for you. On the one hand, your application should satisfy the
+requirements for using this framework. On the other hand, it takes care of a lot
+of stuff for you, so you don't need to worry about everyday problems.
+
+A typical backend does the following:
+
+* Communicates with the frontend (web, mobile) via some wire format (JSON,
+  Protocol Buffers, etc.)
+* Talks to the database
+* Authenticates and authorizes users
+* Has some background jobs
+
+`cake-slayer` works exceptionally well if your application uses:
+
+1. PostgreSQL as database.
+2. Elm on the frontend.
+
+The following Haskell packages have been chosen to provide the necessary
+functional:
+
+* [`postgresql-simple`](https://hackage.haskell.org/package/postgresql-simple),
+  [`postgresql-simple-named`](https://hackage.haskell.org/package/postgresql-simple-named),
+  [`postgresql-simple-migration`](https://hackage.haskell.org/package/postgresql-simple-migration)
+  + For connecting and talking to the PostgreSQL database
+* [`servant`](http://hackage.haskell.org/package/servant),
+  [`servant-swagger`](http://hackage.haskell.org/package/servant-swagger),
+  [`swagger2`](http://hackage.haskell.org/package/swagger2)
+  + A family of libraries for defining and documenting Rest API using type-level eDSL
+* [`elm-street`](http://hackage.haskell.org/package/elm-street)
+  + The bridge between Elm and Haskell — generating Elm data types, JSON
+    encoders and decoders from Haskell types automatically
+* [`jwt`](http://hackage.haskell.org/package/jwt)
+  + User authentication via JWT
+* [`bcrypt`](http://hackage.haskell.org/package/bcrypt)
+  + Secure password hashing
+* [`ekg`](http://hackage.haskell.org/package/ekg),
+  [`prometheus-client`](https://hackage.haskell.org/package/prometheus-client)
+  + Application performance monitoring.
```

Comment:
Let's say that this is absolutely optional and doesn't required to be used (i'm about `prometheus`)